pkg/api/osa/vlabs/convert.go

```diff
@@ -0,0 +1,65 @@
+package vlabs
+
+import (
+	"github.com/Azure/acs-engine/pkg/api"
+)
+
+// AsContainerService returns an OpenShiftCluster converted to an
+// api.ContainerService.
+func (oc *OpenShiftCluster) AsContainerService() *api.ContainerService {
+	cs := &api.ContainerService{
+		ID:       oc.ID,
+		Location: oc.Location,
+		Name:     oc.Name,
+		Tags:     oc.Tags,
+		Type:     oc.Type,
+		Properties: &api.Properties{
+			ProvisioningState: api.ProvisioningState(oc.Properties.ProvisioningState),
```

Comment:
Remind me why we need this? A comment would be nice.

Reply:
I would rather see this as a status struct (that also has condition fields) but this is the way other apis are reporting status.  I assume that there is reporting somewhere based on it.  Maybe @amanohar @qike-ms @nicolerenee  can tell us more

Reply:
@kargakis @pweil- here is more context about ARM's `provisioningState` field: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property

Detailed error is returned by RP to ARM via operation resource: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property